src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -681,6 +682,23 @@ private static IPackageFile FindFileInPackage(string filePath, IEnumerable<IPack
             return null;
         }
 
+        private void ValidateFilesUnique(IEnumerable<IPackageFile> files)
+        {
+            var set = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            foreach (var file in files.Where(t => t.Path != null).Select(t => PathUtility.GetPathWithDirectorySeparator(t.Path)))
+            {
+                if (set.Contains(file))
```

Comment:
Maybe this one is bit performant.
if(!set.Add(file))
{
                    throw new PackagingException(NuGetLogCode.NU5133, string.Format(CultureInfo.CurrentCulture, NuGetResources.FoundDuplicateFile, file));
}

Reply:
Oh cool! Thanks for the heads-up. :)